app/hooks/StrapiAPI.ts

```diff
@@ -262,12 +261,12 @@ export namespace StrapiApi {
     };
   }
 
-  interface DateResponse {
+  export interface CalendarEventResponse {
     id: number;
     startdate: string;
-    enddate: string;
-    starttime: string;
-    endtime: string;
+    enddate: string | null;
+    starttime: string | null;
+    endtime: string | null;
```

Comment:
Let's try our best to be fastidious about matching optionality at the database level.